INTERNALERROR> AssertionError: Test coverage should never be `None`

When running tests in parallel |(for `-n` of one or more), I get this weird `internalerror` message. However, if I run the tests without the `-n 1` option, everything is fine.

```
; py.test --cov=wireless_id_label_editor -n 1 tests/
============================================================================== test session starts ==============================================================================
platform linux2 -- Python 2.7.5, pytest-2.9.2, py-1.4.31, pluggy-0.3.1
rootdir: /home/usr/repos/wireless_id_label_editor, inifile: setup.cfg
plugins: pylama-7.0.9, bdd-2.17.0, catchlog-1.2.2, colordots-0.1, cov-2.3.0, html-1.9.0, pep8-1.0.6, xdist-1.14
gw0 [123]
scheduling tests via LoadScheduling
...........................................................................................................................INTERNALERROR> Traceback (most recent call last):
INTERNALERROR>   File "/home/usr/.virtualenvs/widle/lib/python2.7/site-packages/_pytest/main.py", line 94, in wrap_session
INTERNALERROR>     session.exitstatus = doit(config, session) or 0
INTERNALERROR>   File "/home/usr/.virtualenvs/widle/lib/python2.7/site-packages/_pytest/main.py", line 125, in _main
INTERNALERROR>     config.hook.pytest_runtestloop(session=session)
INTERNALERROR>   File "/home/usr/.virtualenvs/widle/lib/python2.7/site-packages/_pytest/vendored_packages/pluggy.py", line 724, in __call__
INTERNALERROR>     return self._hookexec(self, self._nonwrappers + self._wrappers, kwargs)
INTERNALERROR>   File "/home/usr/.virtualenvs/widle/lib/python2.7/site-packages/_pytest/vendored_packages/pluggy.py", line 338, in _hookexec
INTERNALERROR>     return self._inner_hookexec(hook, methods, kwargs)
INTERNALERROR>   File "/home/usr/.virtualenvs/widle/lib/python2.7/site-packages/_pytest/vendored_packages/pluggy.py", line 333, in <lambda>
INTERNALERROR>     _MultiCall(methods, kwargs, hook.spec_opts).execute()
INTERNALERROR>   File "/home/usr/.virtualenvs/widle/lib/python2.7/site-packages/_pytest/vendored_packages/pluggy.py", line 595, in execute
INTERNALERROR>     return _wrapped_call(hook_impl.function(*args), self.execute)
INTERNALERROR>   File "/home/usr/.virtualenvs/widle/lib/python2.7/site-packages/_pytest/vendored_packages/pluggy.py", line 249, in _wrapped_call
INTERNALERROR>     wrap_controller.send(call_outcome)
INTERNALERROR>   File "/home/usr/.virtualenvs/widle/lib/python2.7/site-packages/pytest_cov/plugin.py", line 221, in pytest_runtestloop
INTERNALERROR>     assert self.cov_total is not None, 'Test coverage should never be `None`'
INTERNALERROR> AssertionError: Test coverage should never be `None`
Coverage.py warning: No data was collected.


---------- coverage: platform linux2, python 2.7.5-final-0 -----------
Name                                              Stmts   Miss Branch BrPart    Cover   Missing
-----------------------------------------------------------------------------------------------
wireless_id_label_editor/RFC2217_client.py           87      0     12      0   100.0%
wireless_id_label_editor/from_reader.py              84      0     22      0   100.0%
wireless_id_label_editor/function_dispatcher.py      22      0      4      0   100.0%
wireless_id_label_editor/gui.py                     276      0     58      0   100.0%
wireless_id_label_editor/nifty_logger.py             56      0      2      0   100.0%
wireless_id_label_editor/reader.py                   13      0      0      0   100.0%
wireless_id_label_editor/to_reader.py                53      0      0      0   100.0%
-----------------------------------------------------------------------------------------------
TOTAL                                               591      0     98      0   100.0%

```

Same thing happens locally with `pytest-cov==2.3.0`.
I've "fixed" it by pinning in tox `pytest-cov==2.2.1`

`INTERNALERROR> Traceback (most recent call last):
INTERNALERROR>   File "/Users/blink/workspace/flowy/.tox/py35/lib/python3.5/site-packages/_pytest/main.py", line 94, in wrap_session
INTERNALERROR>     session.exitstatus = doit(config, session) or 0
INTERNALERROR>   File "/Users/blink/workspace/flowy/.tox/py35/lib/python3.5/site-packages/_pytest/main.py", line 125, in _main
INTERNALERROR>     config.hook.pytest_runtestloop(session=session)
INTERNALERROR>   File "/Users/blink/workspace/flowy/.tox/py35/lib/python3.5/site-packages/_pytest/vendored_packages/pluggy.py", line 724, in __call__
INTERNALERROR>     return self._hookexec(self, self._nonwrappers + self._wrappers, kwargs)
INTERNALERROR>   File "/Users/blink/workspace/flowy/.tox/py35/lib/python3.5/site-packages/_pytest/vendored_packages/pluggy.py", line 338, in _hookexec
INTERNALERROR>     return self._inner_hookexec(hook, methods, kwargs)
INTERNALERROR>   File "/Users/blink/workspace/flowy/.tox/py35/lib/python3.5/site-packages/_pytest/vendored_packages/pluggy.py", line 333, in <lambda>
INTERNALERROR>     _MultiCall(methods, kwargs, hook.spec_opts).execute()
INTERNALERROR>   File "/Users/blink/workspace/flowy/.tox/py35/lib/python3.5/site-packages/_pytest/vendored_packages/pluggy.py", line 595, in execute
INTERNALERROR>     return _wrapped_call(hook_impl.function(*args), self.execute)
INTERNALERROR>   File "/Users/blink/workspace/flowy/.tox/py35/lib/python3.5/site-packages/_pytest/vendored_packages/pluggy.py", line 249, in _wrapped_call
INTERNALERROR>     wrap_controller.send(call_outcome)
INTERNALERROR>   File "/Users/blink/workspace/flowy/.tox/py35/lib/python3.5/site-packages/pytest_cov/plugin.py", line 221, in pytest_runtestloop
INTERNALERROR>     assert self.cov_total is not None, 'Test coverage should never be `None`'
INTERNALERROR> AssertionError: Test coverage should never be `None`
INTERNALERROR> Traceback (most recent call last):
INTERNALERROR>   File "/Users/blink/workspace/flowy/.tox/py35/lib/python3.5/site-packages/_pytest/main.py", line 94, in wrap_session
INTERNALERROR>     session.exitstatus = doit(config, session) or 0
INTERNALERROR>   File "/Users/blink/workspace/flowy/.tox/py35/lib/python3.5/site-packages/_pytest/main.py", line 125, in _main
INTERNALERROR>     config.hook.pytest_runtestloop(session=session)
INTERNALERROR>   File "/Users/blink/workspace/flowy/.tox/py35/lib/python3.5/site-packages/_pytest/vendored_packages/pluggy.py", line 724, in __call__
INTERNALERROR>     return self._hookexec(self, self._nonwrappers + self._wrappers, kwargs)
INTERNALERROR>   File "/Users/blink/workspace/flowy/.tox/py35/lib/python3.5/site-packages/_pytest/vendored_packages/pluggy.py", line 338, in _hookexec
INTERNALERROR>     return self._inner_hookexec(hook, methods, kwargs)
INTERNALERROR>   File "/Users/blink/workspace/flowy/.tox/py35/lib/python3.5/site-packages/_pytest/vendored_packages/pluggy.py", line 333, in <lambda>
INTERNALERROR>     _MultiCall(methods, kwargs, hook.spec_opts).execute()
INTERNALERROR>   File "/Users/blink/workspace/flowy/.tox/py35/lib/python3.5/site-packages/_pytest/vendored_packages/pluggy.py", line 595, in execute
INTERNALERROR>     return _wrapped_call(hook_impl.function(*args), self.execute)
INTERNALERROR>   File "/Users/blink/workspace/flowy/.tox/py35/lib/python3.5/site-packages/_pytest/vendored_packages/pluggy.py", line 249, in _wrapped_call
INTERNALERROR>     wrap_controller.send(call_outcome)
INTERNALERROR>   File "/Users/blink/workspace/flowy/.tox/py35/lib/python3.5/site-packages/pytest_cov/plugin.py", line 221, in pytest_runtestloop
INTERNALERROR>     assert self.cov_total is not None, 'Test coverage should never be `None`'
INTERNALERROR> AssertionError: Test coverage should never be `None`
FINTERNALERROR> Traceback (most recent call last):
INTERNALERROR>   File "/Users/blink/workspace/flowy/.tox/py35/lib/python3.5/site-packages/_pytest/main.py", line 94, in wrap_session
INTERNALERROR>     session.exitstatus = doit(config, session) or 0
INTERNALERROR>   File "/Users/blink/workspace/flowy/.tox/py35/lib/python3.5/site-packages/_pytest/main.py", line 125, in _main
INTERNALERROR>     config.hook.pytest_runtestloop(session=session)
INTERNALERROR>   File "/Users/blink/workspace/flowy/.tox/py35/lib/python3.5/site-packages/_pytest/vendored_packages/pluggy.py", line 724, in __call__
INTERNALERROR>     return self._hookexec(self, self._nonwrappers + self._wrappers, kwargs)
INTERNALERROR>   File "/Users/blink/workspace/flowy/.tox/py35/lib/python3.5/site-packages/_pytest/vendored_packages/pluggy.py", line 338, in _hookexec
INTERNALERROR>     return self._inner_hookexec(hook, methods, kwargs)
INTERNALERROR>   File "/Users/blink/workspace/flowy/.tox/py35/lib/python3.5/site-packages/_pytest/vendored_packages/pluggy.py", line 333, in <lambda>
INTERNALERROR>     _MultiCall(methods, kwargs, hook.spec_opts).execute()
INTERNALERROR>   File "/Users/blink/workspace/flowy/.tox/py35/lib/python3.5/site-packages/_pytest/vendored_packages/pluggy.py", line 595, in execute
INTERNALERROR>     return _wrapped_call(hook_impl.function(*args), self.execute)
INTERNALERROR>   File "/Users/blink/workspace/flowy/.tox/py35/lib/python3.5/site-packages/_pytest/vendored_packages/pluggy.py", line 249, in _wrapped_call
INTERNALERROR>     wrap_controller.send(call_outcome)
INTERNALERROR>   File "/Users/blink/workspace/flowy/.tox/py35/lib/python3.5/site-packages/pytest_cov/plugin.py", line 221, in pytest_runtestloop
INTERNALERROR>     assert self.cov_total is not None, 'Test coverage should never be `None`'
INTERNALERROR> AssertionError: Test coverage should never be `None`
.INTERNALERROR> Traceback (most recent call last):
INTERNALERROR>   File "/Users/blink/workspace/flowy/.tox/py35/lib/python3.5/site-packages/_pytest/main.py", line 94, in wrap_session
INTERNALERROR>     session.exitstatus = doit(config, session) or 0
INTERNALERROR>   File "/Users/blink/workspace/flowy/.tox/py35/lib/python3.5/site-packages/_pytest/main.py", line 125, in _main
INTERNALERROR>     config.hook.pytest_runtestloop(session=session)
INTERNALERROR>   File "/Users/blink/workspace/flowy/.tox/py35/lib/python3.5/site-packages/_pytest/vendored_packages/pluggy.py", line 724, in __call__
INTERNALERROR>     return self._hookexec(self, self._nonwrappers + self._wrappers, kwargs)
INTERNALERROR>   File "/Users/blink/workspace/flowy/.tox/py35/lib/python3.5/site-packages/_pytest/vendored_packages/pluggy.py", line 338, in _hookexec
INTERNALERROR>     return self._inner_hookexec(hook, methods, kwargs)
INTERNALERROR>   File "/Users/blink/workspace/flowy/.tox/py35/lib/python3.5/site-packages/_pytest/vendored_packages/pluggy.py", line 333, in <lambda>
INTERNALERROR>     _MultiCall(methods, kwargs, hook.spec_opts).execute()
INTERNALERROR>   File "/Users/blink/workspace/flowy/.tox/py35/lib/python3.5/site-packages/_pytest/vendored_packages/pluggy.py", line 595, in execute
INTERNALERROR>     return _wrapped_call(hook_impl.function(*args), self.execute)
INTERNALERROR>   File "/Users/blink/workspace/flowy/.tox/py35/lib/python3.5/site-packages/_pytest/vendored_packages/pluggy.py", line 249, in _wrapped_call
INTERNALERROR>     wrap_controller.send(call_outcome)
INTERNALERROR>   File "/Users/blink/workspace/flowy/.tox/py35/lib/python3.5/site-packages/pytest_cov/plugin.py", line 221, in pytest_runtestloop
INTERNALERROR>     assert self.cov_total is not None, 'Test coverage should never be `None`'
INTERNALERROR> AssertionError: Test coverage should never be `None`
Coverage.py warning: No data was collected.`

--- a/src/pytest_cov/plugin.py
+++ b/src/pytest_cov/plugin.py
@@ -165,11 +165,13 @@ class CovPlugin(object):
         if self.options.cov_fail_under is None and hasattr(cov_config, 'fail_under'):
             self.options.cov_fail_under = cov_config.fail_under
 
+    def _is_slave(self, session):
+        return hasattr(session.config, 'slaveinput')
+
     def pytest_sessionstart(self, session):
         """At session start determine our implementation and delegate to it."""
         self.pid = os.getpid()
-        is_slave = hasattr(session.config, 'slaveinput')
-        if is_slave:
+        if self._is_slave(session):
             nodeid = session.config.slaveinput.get('slaveid',
                                                    getattr(session, 'nodeid'))
             self.start(engine.DistSlave, session.config, nodeid)
@@ -211,7 +213,7 @@ class CovPlugin(object):
         if self.cov_controller is not None:
             self.cov_controller.finish()
 
-        if self._should_report():
+        if not self._is_slave(session) and self._should_report():
             try:
                 self.cov_total = self.cov_controller.summary(self.cov_report)
             except CoverageException as exc:

--- a/tests/test_pytest_cov.py
+++ b/tests/test_pytest_cov.py
@@ -3,6 +3,7 @@ import os
 import subprocess
 import sys
 from distutils.version import StrictVersion
+from itertools import chain
 
 import coverage
 import py
@@ -447,15 +448,17 @@ def test_foo(foo):
     result = testdir.runpytest('-v',
                                '--cov=%s' % script.dirpath(),
                                '--cov-report=term-missing',
-                               '-n', '5',
+                               '-n', '5', '-s',
                                script)
     result.stdout.fnmatch_lines([
         '*- coverage: platform *, python * -*',
         'test_dist_combine_racecondition* 2002 * 0 * 100%*',
         '*1000 passed*'
     ])
-    for line in result.stdout.lines:
+
+    for line in chain(result.stdout.lines, result.stderr.lines):
         assert 'The following slaves failed to return coverage data' not in line
+        assert 'INTERNALERROR' not in line
     assert result.ret == 0
 
 

